### PR TITLE
Mingw port fixes

### PR DESCRIPTION
**Pull request by koprok**
- Fix MinGW build of following ports: openal-soft, glew, protobuf
- Update protobuf to 3.12.3

### CODE REVIEW

**Comment by longnguyen2004** [comment]
Uhh since `mingw` is currently static only so import libs check is already disabled internally in vcpkg (or am I wrong here?)
It would be useful for dynamic triplets tho, which I'm planning to add in #12101

**Comment by koprok** [comment]
> Uhh since `mingw` is currently static only so import libs check is already disabled internally in vcpkg (or am I wrong here?)
> It would be useful for dynamic triplets tho, which I'm planning to add in #12101

@longnguyen2004 you are right, I will remove this commit.

File: ports/protobuf/portfile.cmake [hunk 1]
```diff
@@ -12,7 +12,7 @@ vcpkg_from_github(
 
 if(CMAKE_HOST_WIN32 AND NOT VCPKG_TARGET_ARCHITECTURE MATCHES "x64" AND NOT VCPKG_TARGET_ARCHITECTURE MATCHES "x86")
     set(protobuf_BUILD_PROTOC_BINARIES OFF)
-elseif(CMAKE_HOST_WIN32 AND VCPKG_CMAKE_SYSTEM_NAME)
+elseif(CMAKE_HOST_WIN32 AND VCPKG_TARGET_IS_UWP)
```
Comment (JackBoosY) [inline ports/protobuf/portfile.cmake:15]: `VCPKG_CMAKE_SYSTEM_NAME` is not equal to `VCPKG_TARGET_IS_UWP`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/protobuf/portfile.cmake (hunk 1 above)
Comment (koprok) [reply ports/protobuf/portfile.cmake:15]: You are right, but this change is intentional. `VCPKG_CMAKE_SYSTEM_NAME` is set for both WindowsStore and MinGW targets. I want this branch of the if-else to execute only when target is WindowsStore, but not when it is MinGW.

**Review by JackBoosY — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
I've changed the conditional to better reflect existing behavior -- LGTM once the build passes again :)

**Comment by JackBoosY** [comment]
Passed.

**Review by ras0219-msft — APPROVED** [review]

**Comment by JackBoosY** [comment]
```
D:\buildtrees\magnum-plugins\src\v2020.06-f0c46dc4fa.clean\src\MagnumPlugins\StbImageImporter\StbImageImporter.cpp(91): error C3861: 'stbi_set_flip_vertically_on_load_thread': identifier not found
D:\buildtrees\magnum-plugins\src\v2020.06-f0c46dc4fa.clean\src\MagnumPlugins\StbImageImporter\StbImageImporter.cpp(95): error C3861: 'stbi_convert_iphone_png_to_rgb_thread': identifier not found
```

**Pull request merged**